db/migrate/20160414185708_create_container_label_tag_mappings.rb

```diff
@@ -0,0 +1,13 @@
+class CreateContainerLabelTagMappings < ActiveRecord::Migration[5.0]
+  def change
+    create_table :container_label_tag_mappings do |t|
+      t.string :labeled_resource_type
+      t.string :label_name
+      t.text :label_value
+      t.bigint :category_tag_id
+      t.bigint :tag_id
```

Comment:
@alongoldboim commented on https://github.com/ManageIQ/manageiq/pull/7605#discussion_r60007351

> - I like to use 
>
>   ```
>   t.belongs_to :tag, :type => :bigint
>   ```
>
>   It has no difference but its clearer.

I've considered `t.references` but saw practically none of our migrations used it — but I now see the `t.belongs_to` alias is widespread.
The stopper was that based on some googling I can't use this style with 2 names referring into the same tables [http://stackoverflow.com/a/22384289/239657 is closest but requires an extra gem].
IMHO, mixing:
```
t.belongs_to :tag, :type => :bigint
t.bigint :category_tag_id
```
would be less readable?

> - I think you can drop the "category_tag_id" if the tag is connected to its category, i assume you can get it from the tag (.classifaction/.category) and place a :through ?

That's true for (name, value) -> specific tag mappings.  For these the code only looks at `tag`; `category_tag` doesn't matter and if it did it could be computed from `tag`.

However I also have open-ended (name, *) -> * mappings where the tag is created on demand.
These rows have no `label_value` and no `tag`, but need some way to specify a category under which the tag should be created — that's what `category_tag` does.
[My current implementation does create a specific-tag mapping row every time it creates a specific tag.  But it still needs the open-ended row in case it encounters new values in the future.]

- I could abuse `tag` to double-serve as the parent tag or specific tag depending on whether `label_value` is set.
  I thought it'd be messy.

P.S. Could you please take a look at the comment in app/models/container_label_tag_mapping.rb
 and tell me if they explain this well?